### PR TITLE
fix: fallback to english challenges

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
All challenges will use the english version if a translated file is not available.  SHOW_NEW_CURRICULUM still gates what's shown in the client.

With this it's possible to build i18n clients without needing to populate their `/curriculum` directories with copies of the english.

### CODE REVIEW

**Review by moT01 — DISMISSED** [review]
Well, I looked over this pretty good - more-so did testing, but looked through the code some as well. Everything seems to be working...

- [x] creates i18n challenges when superblock folder is missing
- [x] creates i18n challenges when block folder is missing
- [x] creates i18n challenges when files are missing
- [x] doesn't create i18n challenges when superblock is a "new curriculum" (`SHOW_NEW_CURRICULUM`)

The only thing that might be nice here is some logs - maybe `Did not find <block>/<superblock>/<file> for <language>, using English files.` Not necessary though.

A PR like this makes me a little nervous, but I tested it quite good - so I would be fine adding it.. pending another set or two of eyes.

**Review by naomi-lgbt — DISMISSED** [review]
I've poked at this a bit and everything seems good on my end.

A nice to have thing (that can totally go in a separate PR) would be to have a check where an "isAudited" cert that's missing files throws a warning or something.

**Comment by ojeytonwilliams** [comment]
> A nice to have thing (that can totally go in a separate PR) would be to have a check where an "isAudited" cert that's missing files throws a warning or something.

Yep, that does seem useful.  The way I've implemented it means that, right now, if you try to show upcoming changes for espanol, say, it will fail due to JS and D3 practice projects.

Ideally those should be in their own superblocks, anyway, so I think this is fine, but let me know.

**Review by naomi-lgbt — APPROVED** [review]
> if you try to show upcoming changes for espanol, say, it will fail due to JS and D3 practice projects.

This should be fine - `SHOW_UPCOMING_CHANGES` is never true on production, and we have consistently landed English changes first before landing them on i18n instances.

> Ideally those should be in their own superblocks, anyway, so I think this is fine, but let me know.

They should - we can migrate them in a separate PR.

File: curriculum/getChallenges.js
Comment (moT01) [inline curriculum/getChallenges.js:364]: I took another look at this and did a little playing around to see what lessons were using English. It looks like we use English for anything that isn't audited. So when I build in Espanol, I see all the English challenges for coding interview prep and other sections where many of the challenges are translated. This is the same on production. Is this the behavior we want? Seems like if it exists in the language, we should use it. I dunno - maybe we like to wait so we can have an announcement or something that all the challenges are translated. 

Makes me wonder if there's any sections that are 100% translated, but we're still using the English files because we didn't add them to the isAudited arrays. I took a quick look at those arrays and they're pretty full, so it looks like we've been keeping up with it.

Anyway - I don't think that should hold up this PR since it's the same on production, so I'm going to go ahead with it.

**Review by moT01 — APPROVED** [review]

File: curriculum/getChallenges.js
Comment (nhcarrigan) [reply curriculum/getChallenges.js:364]: That's the intended behaviour, yes - the idea being that we aren't displaying a superblock with only partial translations.

**Pull request merged**
